src/group_theory/subgroup/actions.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2020 Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kexing Ying
+-/
+
+import group_theory.subgroup.basic
+
+/-!
+# Actions by `subgroup`s
```

Comment:
I think it would make sense to make this split across `submonoid`, `subgroup`, `subsemiring` too; perhaps even doing that first, and then rebasing this PR on top of that.

Reply:
These seem to be less monolithic already. I could work on this if you were a little more specific about what those splits would look like, but I'd prefer not blocking on that.